Enforce full distribution of high temp radiant energy to surfaces and people

RKStrand opened the pull request:
## Pull request overview

This pull request includes the code fix, unit test, and minor documentation mod for a problem in the high temperature radiant heater code.  In the code, a warning message was produced when the user did not assign all of the radiant energy emitted from the radiant heater to surfaces or people in a space.  The code noted that the sum did not add up to unity and produced the warning but then simply continued to simulate things.  This resulted in a loss of energy that could be up to 100% of the radiant energy.  This was discovered by a user and a change was request.  Now, the code will produce a severe warning and will also produce additional information telling the user what was processed and that this is not allowed.  The documentation was also modified to clarify this as well.  This fix applies to all high temperature radiant heater types and was logged as GitHub Issue #4324.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This defect is logged as ZoneHVAC:HighTemperatureRadiant - No radiant receiving surfaces and no radiant to people or GitHub Issue #4324.
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes n/a
- [ ] Documentation changes in place

RKStrand committed:
Bug fix and doc update

This commit includes the code to fix this bug and a slight modification
to the documentation in the IO Ref to clarify what EnergyPlus is doing.
 As per discussion with Mike Witte, the error call has been changed to
severe and the user is required to account for all radiant energy
(within a small tolerance).  The error message produced when this does
not happen has also been improved to provide the user with more
information about the problem.

RKStrand committed:
Merge branch 'develop' into 113837787-High-Temp-Radiant-Energy-to-Surface-Imbalance

RKStrand committed:
Unit test

Addition of a unit test for this bug fix.  The unit test shows the new
message when the user fails to account for all radiant energy from the
high temperature radiant heater.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 113837787-High-Temp-Radiant-Energy-to-Surface-Imbalance

mjwitte commented on src/EnergyPlus/HighTempRadiantSystem.cc:
```diff
@@ -219,11 +219,14 @@ namespace HighTempRadiantSystem {
 
 		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
 		static bool GetInputFlag( true ); // First time, input is "gotten"
```
@RKStrand Now that there's a unit test for high temp radiant, this needs to move up to anonymous namespace and there needs to be a HighTempRadiant::clear_state function.  Let me know if you need an example to follow.

RKStrand replied:
@mjwitte Yeah, probably need an example as I am simply following examples from others at this point and I don't always catch the finer points.  Thanks in advance.

mjwitte commented on tst/EnergyPlus/unit/HighTempRadiantSystem.unit.cc:
```diff
@@ -0,0 +1,140 @@
+
+	}
+
+}
```
@RKStrand Need a return at the end of this file to get rid of "no newline" warning in github.  BTW, I've merged in develop, so pull this branch before making changes.

RKStrand replied:
@mjwitte Ok, I'll add in a line return at the end of the file after I pull the branch you put up.

mjwitte commented on doc/input-output-reference/src/overview/group-radiative-convective-units.tex:
```diff
@@ -2063,3 +2069,5 @@ \subsubsection{Inputs}\label{inputs-11-013}
       Z5VentSlabOut;           !- Slab Outlet Node Name for Surface 3
 \end{lstlisting}
+\]
+\end{document}
```
@RKStrand I/O Ref doesn't build.  These two added lines look suspect. I would delete them.

Myoldmopar replied:
Yeah, @RKStrand adds in this and the above preamble section so that he can build this chunk of documentation standalone.  Works well, but yeah, it'll have to go before merge.

RKStrand replied:
@mjwitte @Myoldmopar Well, again as with another one, I'm not putting these extra things in so there must be something with the software I am using that is putting this stuff in without my knowledge.  I am using LaTeXiT on the Mac.  Any ideas why this is happening?

mjwitte replied:
LaTexiT is probably adding it.  There may be an option to prevent it or you may need to diff the file after editing and remove any extra changes like this.

mjwitte commented on doc/input-output-reference/src/overview/group-radiative-convective-units.tex:
```diff
@@ -1,4 +1,10 @@
+\usepackage{amssymb} %maths
+\usepackage{amsmath} %maths
+\usepackage[utf8]{inputenc} %useful to type directly diacritic characters
+\begin{document}
```
These six lines are the ones @Myoldmopar referred to as preamble.  Just delete theses and the other extra lines in a text editor and the docs should be OK then.

RKStrand committed:
Corrected issues noted by reviewer

Made additional changes to the code to address comments made by
@mjwitte during his review of this bug fix.  This hopefully resolves
all of the issues he noted.  After this commit, another commit to
resolve any merge conflicts will need to be made.

RKStrand committed:
Merge branch 'develop' into 113837787-High-Temp-Radiant-Energy-to-Surface-Imbalance

# Conflicts:
#	tst/EnergyPlus/unit/CMakeLists.txt

RKStrand commented:
@mjwitte Ok, I have added a variety of things to address your comments above.  I think that takes care of everything?  Also, I'm not sure why it is saying I have branch conflict.  The file with a "conflict" was the CMake file in the tst folder...which I had added something to but didn't see any other changes to it.  Am I missing something else?

mjwitte commented:
That type of conflict usually means that you've added a line to the file at the same place that another merge into develop added a line.  If you look at that file in develop and compare with the one in your branch you should see it.

RKStrand commented:
@mjwitte That's what I thought too.  First, while in the new branch, I copied the CMake file to my desktop and changed its name.  Then, I downloaded the latest version of develop and copied the CMake file to my desktop.  I diffed the files and the only difference is the line that I added.  Nothing missing or overwritten.  Did Git Desktop mislead me as far as where there was a conflict?  It seemed pretty clear that the conflict was in this file since it named it specifically...

mjwitte commented:
@RKStrand That can happen if your branch adds the same line that was added in another branch.  But looking at the diffs on github for this branch, something is very wrong.  It's showing over 700 files changed and showing additions that have nothing to do with this pull request.  We may need @Myoldmopar to help sort out the problem.

Myoldmopar commented:
I'm trembling with fear of line endings and white space... OK not
trembling, and not afraid, but feels suspicious. Tomorrow morning I'll look
unless i get some extra energy tonight.

nrel-bot-3 commented:
@RKStrand @lgentile it has been 14 days since this pull request was last updated.

Myoldmopar commented:
@RKStrand This is bad. And I'm not sure how it happened.  There are merges, and merge conflicts, in so many files.  There's no reason there should be so many merges.  And I don't think there is just a line ending problem, there are lines going in and out.  What are the chances you could just pull develop down fresh, and copy in only the files with the bug fix?

RKStrand commented:
@Myoldmopar Yeah, I have literally no idea what happened.  I know it looked pretty crazy, but I thought that was because a whole bunch of merges were done.  I guess the best thing I can do is try to do this from scratch based on the changes I made.  I'm not sure the best way to do this, but I will try to work on it this afternoon.

RKStrand commented:
@Myoldmopar So, should I just start a new branch based on develop?

Myoldmopar commented:
Yeah, if you can just checkout develop, pull, and then copy in your
changes, that should be it.

mjwitte commented:
@RKStrand Yes, I would.  New branch, new pull request.  You can look at the individual commits in this PR on github to find the originally changed code and reproduce the changes or copy those lines out of the source files in this branch.

RKStrand commented:
@Myoldmopar @mjwitte Thanks, Gentlemen...I will try to get working on that as soon as possible based on what you both said.  That was kinda my impression also as the best way forward but I thought I would double check first.

mjwitte commented:
Replaced by #5830.
